projects/astarfarm/index.js

```diff
@@ -0,0 +1,16 @@
+const ethers = require("ethers")
+
+const fetch = async () => {
+    const rpcUrl = "https://rpc.astar.network:8545";
+    const provider = await ethers.getDefaultProvider(rpcUrl);
+    const contract = new ethers.Contract(
+      "0x0BdD0C5406B9420B4d4251671F9F045e4FBEb758",
+      ["function staked() external view returns (uint128)"],
+      provider
+    );
+    return await contract.staked(); // tvl as wei
+}
+
+module.exports = {
+    fetch
+}
```

Comment:
Can you change the adapter to this:

    const sdk = require('@defillama/sdk')

    const stakingContract = '0x0BdD0C5406B9420B4d4251671F9F045e4FBEb758'

    module.exports = {
        astar: {
          tvl: async (ts, _block, { astar: block }) => {
            const { output: lockedAstar } = await sdk.api.eth.getBalance({ target: stakingContract, block, chain: 'astar'})
            return { 'astar': +lockedAstar / 1e18 }
          }
        }
    }

Reply:
I asking for the change because I think the fetch adapter would fail because it would be return value in astar instead of usd right?
Also, we might want to use different rpc in the future, with my adapter, it is easy for us to change, also fetch historical data